### PR TITLE
Adds RockDrop spell

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Adds a new wizard spell, `Rockdrop`. The final spell for #5160.
balderdash
- Converts `Lesser Summon Guns` to use new portal spawning method.
    - Should fix issue of portal and gun animations not working on clients.
- New component `RandomInitialSprite`. Simply sets the SpriteHandler to use a random SO from the list at initialisation.

CL:Added RockDrop spell.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Sprite Handler/RandomInitialSprite.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace Core.Sprites
+{
+	/// <summary>
+	/// Initializes the object with a random sprite SO from the SO list defined in SpriteHandler component.
+	/// </summary>
+	public class RandomInitialSprite : MonoBehaviour
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/RandomInitialSprite.cs]: you could probably include this as part of Sprite handler may be?

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/RandomInitialSprite.cs]: Done; please confirm changes to `SpriteHandler` look good.

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 2]
```diff
@@ -22,6 +22,8 @@ public class SpriteHandler : MonoBehaviour
 	[SerializeField] private SpriteDataSO PresentSpriteSet;
 	private SpriteDataSO.Frame PresentFrame = null;
 
+	[SerializeField] private bool randomInitialSprite = false;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:26]: add tooltip explaining what it does

File: UnityProject/Assets/Scripts/Items/Food/Cigarette.cs [hunk 3]
```diff
@@ -135,7 +135,7 @@ private void Burn()
 	{
 		var worldPos = gameObject.AssumedWorldPosServer();
 		var tr = gameObject.transform.parent;
-		var rotation = RandomUtils.RandomRotatation2D();
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Food/Cigarette.cs:138]: rotatatatata

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs (hunk 2 above)
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:26]:
```suggestion
	[Tooltip("If checked, a random sprite SO will be selected during initialization from the catalogue of sprite SOs.")]
	[SerializeField] private bool randomInitialSprite = false;
```

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:26]: Done; see https://github.com/unitystation/unitystation/pull/5766/files/682965c38e233ed920f8aecd20360a16b2607e60#diff-ebae75d08489df0a3c75ab87b5b8d5cdd7551b791075ce40301e9bcfbeecfbe8

File: UnityProject/Assets/Scripts/Items/Food/Cigarette.cs (hunk 3 above)
Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Food/Cigarette.cs:138]: Rotatata indeed.

**Pull request merged**